registry/auth.go

```diff
@@ -36,6 +37,73 @@ type ConfigFile struct {
 	rootPath string
 }
 
+type RequestAuthorization struct {
+	authConfig       *AuthConfig
+	registryEndpoint *Endpoint
+	resource         string
+	scope            string
+	actions          []string
+}
+
+func NewRequestAuthorization(authConfig *AuthConfig, registryEndpoint *Endpoint, resource, scope string, actions []string) *RequestAuthorization {
+	return &RequestAuthorization{
+		authConfig:       authConfig,
+		registryEndpoint: registryEndpoint,
+		resource:         resource,
+		scope:            scope,
+		actions:          actions,
+	}
+}
+
+func (auth *RequestAuthorization) getToken() (string, error) {
```

Comment:
The TODOs involve optimizing when a token is retrieved.  Because the tokens are short lived, they should not be cached indefinitely.  The optimization is cache for a short period of time.  The implication of not caching is a new token is retrieved for every registry request, which will be two per layer + a call for the manifest.